src/tests/build.proj

```diff
@@ -592,6 +670,11 @@
       <GroupBuildCmd>$(GroupBuildCmd) /bl:$(ArtifactsDir)/log/$(Configuration)/InnerManagedTestBuild.$(__TestGroupToBuild).binlog</GroupBuildCmd>
       <GroupBuildCmd Condition="'$(CrossBuild)' == 'true'">$(GroupBuildCmd) "/p:CrossBuild=true"</GroupBuildCmd>
       <GroupBuildCmd Condition="'$(TestBuildMode)' == 'nativeaot'">$(GroupBuildCmd) "/p:DefaultBuildAllTarget=BuildNativeAot"</GroupBuildCmd>
+      <GroupBuildCmd Condition="'$(TestBuildMode)' == 'nativeaot' and '$(TargetsAppleMobile)' == 'true'">$(GroupBuildCmd) "/p:UseNativeAOTRuntime=true"</GroupBuildCmd>
+      <GroupBuildCmd Condition="'$(TestBuildMode)' == 'nativeaot' and '$(TargetsAppleMobile)' == 'true'">$(GroupBuildCmd) "/p:NativeLib=static"</GroupBuildCmd>
+      <GroupBuildCmd Condition="'$(TestBuildMode)' == 'nativeaot' and '$(TargetsAppleMobile)' == 'true'">$(GroupBuildCmd) "/p:CustomNativeMain=true"</GroupBuildCmd>
+      <GroupBuildCmd Condition="'$(TestBuildMode)' == 'nativeaot' and '$(TargetsAppleMobile)' == 'true'">$(GroupBuildCmd) "/p:NativeCompilationDuringPublish=false"</GroupBuildCmd>
```

Comment:
Why are we setting `NativeCompilationDuringPublish` to false?

Reply:
It is used to escape the Native AOT bundling targets as they report `NativeLib requires OutputType=Library`. If we skip the error when the CustomNativeMain is used by adding the `'$(CustomNativeMain)' != 'true'` condition to the 

```<Error Condition="'$(OutputType)' != 'Library' and '$(NativeLib)' != ''" Text="NativeLib requires OutputType=Library." />```

 it should work properly and there is no need to avoid the publish targets.

Reply:
@ivanpovazan re: https://github.com/dotnet/runtime/issues/87187#issuecomment-1580202584 Target `LinkNative` is probably something we don't need on iOS and would like to exclude?

Reply:
Correct. In the initial setup for the sample HelloiOS app and the AppleAppBundler we tried to escape what we do not need from `Microsoft.NETCore.Native.Publish.targets`, `LinkNative` being one of the targets we want to overwrite and perform it as part of the "host" build/bundle process. These exclusions and workarounds are also true for the current Xamarin setup.

I tried to list all the "problematic" properties that should be cleaned up (and simplified) in: https://github.com/dotnet/runtime/issues/87187